kubernetes/samples/scripts/create-rcu-schema/create-rcu-schema.sh

```diff
@@ -56,38 +72,61 @@ if [ -z ${rcuType} ]; then
 fi
 
 if [ -z ${pullsecret} ]; then
-  pullsecret="docker-store"
+  pullsecret="none"
+  pullsecretPrefix="#"
 fi
 
 if [ -z ${fmwimage} ]; then
  fmwimage="container-registry.oracle.com/middleware/fmw-infrastructure:12.2.1.4"
 fi
 
+if [ -z ${namespace} ]; then
+ namespace="default"
+fi
+
+if [ -z ${sysPassword} ]; then
+ sysPassword="Oradoc_db1"
+fi
+
+if [ -z ${schemaPassword} ]; then
+ schemaPassword="Oradoc_db1"
+fi
+
+if [ -z ${rcuOutputDir} ]; then
+ rcuOutputDir="rcuoutput"
+fi
+
 echo "ImagePullSecret[$pullsecret] Image[${fmwimage}] dburl[${dburl}] rcuType[${rcuType}]"
 
+mkdir -p ${rcuOutputDir}
+rcuYaml=${rcuOutputDir}/rcu.yaml
+rm -f ${rcuYaml}
+rcuYamlTemp=${scriptDir}/common/template/rcu.yaml.template
+cp $rcuYamlTemp $rcuYaml
+
 #kubectl run rcu --generator=run-pod/v1 --image ${jrf_image} -- sleep infinity
 # Modify the ImagePullSecret based on input
-sed -i -e '$d' ${scriptDir}/common/rcu.yaml
-echo '           - name: docker-store' >> ${scriptDir}/common/rcu.yaml
-sed -i -e "s?name: docker-store?name: ${pullsecret}?g" ${scriptDir}/common/rcu.yaml
-sed -i -e "s?image:.*?image: ${fmwimage}?g" ${scriptDir}/common/rcu.yaml
-kubectl apply -f ${scriptDir}/common/rcu.yaml
+sed -i -e "s:%NAMESPACE%:${namespace}:g" $rcuYaml
+sed -i -e "s:%WEBLOGIC_IMAGE_PULL_SECRET_NAME%:${pullsecret}:g" $rcuYaml
+sed -i -e "s:%WEBLOGIC_IMAGE_PULL_SECRET_PREFIX%:${pullsecretPrefix}:g" $rcuYaml
+sed -i -e "s?image:.*?image: ${fmwimage}?g" $rcuYaml
+kubectl apply -f $rcuYaml
 
 # Make sure the rcu deployment Pod is RUNNING
-checkPod rcu default
-checkPodState rcu default "1/1"
+checkPod rcu $namespace
+checkPodState rcu $namespace "1/1"
 sleep 5
-kubectl get po/rcu 
+kubectl get po/rcu -n $namespace 
 
 # Generate the default password files for rcu command
-echo "Oradoc_db1" > pwd.txt
-echo "Oradoc_db1" >> pwd.txt
+echo "$sysPassword" > pwd.txt
+echo "$schemaPassword" >> pwd.txt
 
-kubectl exec -i rcu -- bash -c 'cat > /u01/oracle/createRepository.sh' < ${scriptDir}/common/createRepository.sh 
-kubectl exec -i rcu -- bash -c 'cat > /u01/oracle/pwd.txt' < pwd.txt 
+kubectl -n $namespace exec -i rcu -- bash -c 'cat > /u01/oracle/createRepository.sh' < ${scriptDir}/common/createRepository.sh 
```

Comment:
Can we please move the "-n $namespace" after "exec"?  Then I'll still be able to grep our code base for every use of "kubectl exec".  There are changes coming to kubectl, including changes to the "kubectl exec" format.

Reply:
Hi @rjeberhard ,

Incorporated your comments into the latest commit.
Please review.

Thanks,
Asha